### PR TITLE
Fix Bug 1221087, set map state correctly for sub page landing state

### PR DESCRIPTION
**Pull request by schalkneethling**
@alexgibson r?

### CODE REVIEW

File: media/js/mozorg/contact-spaces.js [hunk 1]
```diff
@@ -408,6 +409,7 @@
             var state = mozMap.getMapState();
             if (state === 'spaces') {
                 initContentId = $('#nav-spaces li.current').data('id');
+                mozMap.showSpace(state.url, null);
```
Comment (alexgibson) [inline media/js/mozorg/contact-spaces.js]: Looks like this causes a redundant ajax request, would it be better just to show the correct marker/map position here instead?

File: media/js/mozorg/contact-spaces.js [hunk 2]
```diff
@@ -416,6 +418,7 @@
                 }
             } else if (state === 'communities') {
                 initContentId = $('#nav-communities li.current').data('id');
+                mozMap.setMapState();
```
Comment (alexgibson) [inline media/js/mozorg/contact-spaces.js]: Calling this here doesn't fix the bug for displaying the community map overlay?

File: media/js/mozorg/contact-spaces.js [hunk 3]
```diff
@@ -160,6 +160,7 @@
                     // Hide spaces and community nav+meta
                     mozMap.toggleNav('contact');
                     mozMap.toggleMobileNav('contact');
+                    mozMap.setMapState();
```
Comment (alexgibson) [inline media/js/mozorg/contact-spaces.js]: Does this need to be called here? Looking at the code, it seems like this should only need calling if the active tab changes (see line 137)

Comment (schalkneethling) [inline media/js/mozorg/contact-spaces.js]: Yeah, it seemed like when you switch from a specific space, for example, back to contact us, the map does not change. Perhaps it is just loading really slowly for me?

File: media/js/mozorg/contact-spaces.js (hunk 2 above)
Comment (schalkneethling) [inline media/js/mozorg/contact-spaces.js]: Hmmm, it worked for me, although it does take a looong time to load. Will test some more.

File: media/js/mozorg/contact-spaces.js (hunk 1 above)
Comment (schalkneethling) [inline media/js/mozorg/contact-spaces.js]: Of course, I have it at the line right above this one :blush:

File: media/js/mozorg/contact-spaces.js (hunk 3 above)
Comment (alexgibson) [inline media/js/mozorg/contact-spaces.js]: Hmm, I wonder if this is just masking over another bug perhaps - something that was added with the contact tab?

File: media/js/mozorg/contact-spaces.js (hunk 2 above)
Comment (alexgibson) [inline media/js/mozorg/contact-spaces.js]: I think here you might want to use something like:

    var region = initContentId ? initContentId : 'communities';
mozMap.showCommunityRegion

**Comment by schalkneethling** [comment]
@alexgibson Ok, I have made some updates and, for me, it seems like all the kinks are sorted now.

File: media/js/mozorg/contact-spaces.js [hunk 4]
```diff
@@ -408,6 +409,7 @@
             var state = mozMap.getMapState();
             if (state === 'spaces') {
                 initContentId = $('#nav-spaces li.current').data('id');
+                mozMap.showSpace(state.url, initContentId);
```
Comment (alexgibson) [inline media/js/mozorg/contact-spaces.js]: This is still causing a redundant ajax request on page load. `mozMap.showSpace()` does a lot more than setting the map looking at the code. I seem to remember addressing this ajax issue was how the regression snuck in to begin with.

File: media/js/mozorg/contact-spaces.js (hunk 3 above)
Comment (alexgibson) [inline media/js/mozorg/contact-spaces.js]: This still doesn't fix this issue you describe when I test locally (switch from a specific space, for example, back to contact us).

Comment (schalkneethling) [inline media/js/mozorg/contact-spaces.js]: Hmmmm, so I only see requests being made for spaces when it has not already been loaded. Are you seeing it always make a request to load the content or, are you referring to a different request?

File: media/js/mozorg/contact-spaces.js (hunk 4 above)
Comment (schalkneethling) [inline media/js/mozorg/contact-spaces.js]: Hmmmm, so I only see requests being made for spaces when it has not already been loaded. Are you seeing it always make a request to load the content or, are you referring to a different request?

Comment (alexgibson) [inline media/js/mozorg/contact-spaces.js]: Hard-refresh on any Space URL and I see the same content load twice

File: media/js/mozorg/contact-spaces.js (hunk 3 above)
Comment (schalkneethling) [inline media/js/mozorg/contact-spaces.js]: Hmmmmm, this is wacky. It seems like it is very unstable. Let me continue looking at it. Do we have functional tests for this page?

Comment (alexgibson) [inline media/js/mozorg/contact-spaces.js]: Yes we have functional tests - I don't think it is unstable per-say, just something isn't being set quite right. The contact tab should need to follow the same logic as the other two tabs.

**Comment by schalkneethling** [comment]
@alexgibson ok, I have updated the PR. From what I can see, this works well. If you are seeing problems though or, redundant Ajax calls, please let me know.

File: media/js/mozorg/contact-spaces.js [hunk 5]
```diff
@@ -478,8 +490,6 @@
                 mozMap.hideMapLegend();
                 // reposition markers above the labels
                 mozMap.setLabelLayerIndex(1);
-
-                mozMap.updateTabState('contact');
```
Comment (alexgibson) [inline media/js/mozorg/contact-spaces.js]: Not introduced in this change, but here if `state === 'contact'` is true, we're calling `mozMap.unbindCommunityNav();` to unbind the communities navigation, but we're seemingly not not also calling `mozMap.unbindSpacesNav();`

**Comment by alexgibson** [comment]
r+wc

**Comment by alexgibson** [comment]
Looks good, thanks @schalkneethling!

**Pull request merged**